### ISSUE TITLE
Facilitate gradient interoperability between other ml frameworks (TensorFlow, Torch, etc) and jax

### ISSUE THREAD

**Issue by phinate** [pyhf/neos#3]
To use the setup as it is now, one would need to define their neural network (or whatever parametric observable you like that yields histograms) using only jax, which allows the tracing of gradients. This is generally seen as a limitation for two reasons:

- New users will not want to learn *yet another* way to do neural networks
- Model implementations that are non-trivially portable between frameworks have no way to link with the functionality of the library

[This issue](https://github.com/google/jax/issues/2154) outlines the kind of syntax and operations needed to enable this, but jax is still in it's infancy, so there is no good user-facing API for this yet. It seems like it could be possible! (:

**Comment by lukasheinrich** [comment]
we have an example here for pytorch using metadifff

![Uploading image.png…]()

**Comment by phinate** [comment]
> we have an example here for pytorch using metadifff
> 
> 
> 
> ![Uploading image.png…]()
> 
> 

Awesome, but I think that you've just posted a blank markdown link there :P
